src/upload.rs

```diff
@@ -205,4 +217,72 @@ impl<'a> Uploader<'a> {
             .map_err(|e| Error::RequestError(Box::new(e)))?
             .into_string()?)
     }
+
+    /// Retrieves and prints the files on server.
+    pub fn retrieve_list(&self, prettify: bool) -> Result<()> {
```

Comment:
I would say let's make this function take a generic argument `<Output: Write>` and replace the `println!` statements with `writeln!`. For example:

```rs
    pub fn retrieve_list<Output: std::io::Write>(&self, output: &mut Output, prettify: bool) -> Result<()> {
       
        writeln!(output, "...")?;
     
        Ok(())
    }
```

This will make the code testable since it takes a type which implements `Write`. This means that we can use any type (such as `Vec<u8>`) to mock the function in unit tests and assert the output.

Reply:
Hmm, I am not entirely sure what that means. So output is basically a reference that is returned from the function, correct?

I have to look into this. Currently it makes no sense to me.

Reply:
> So output is basically a reference that is returned from the function, correct?

Not quite. `Output` is the parameter of the function which can be any type that implements `Write` trait.

Reply:
Ok, so how do I call the function from `lib.rs` then? No worries, I'll figure it out. 

I never thought about testing since the cli code does not have any test cases.

Reply:
done

Reply:
Oops, I shouldn't resolve them. This is usually done by the person who started the thread. Sorry. Unresolving.